docs/guide/tutorials/aggregation.txt

```diff
@@ -1,3 +1,137 @@
+sample dataset to find a list of restaurants located in the Bronx,
+grouped by restaurant category.
+
+.. code-block:: js
+
+   const { MongoClient } = require('mongodb');
+   
+   // Connection URL
+   const url = 'mongodb://localhost:27017';
+   
+   // Create a new MongoClient
+   const client = new MongoClient(url);
+
+   async function main(client) {
+     const collection = client.db('myproject').collection('restaurants');
+
+     const cursor = collection.aggregate([
+       { $match: { borough: 'Bronx' } },
+       { $unwind: '$categories'},
+       { $group: { _id: '$categories', 'Bronx restaurants': { $sum: 1 } } }
+     ]);
+
+     const documents = await cursor.toArray();
+     console.log(documents);
+   }
+   
+   // Function to connect to the server and run your code
+   async function run() {
+     try {
+       // Connect the client to the server
+       await client.connect();
+       console.log('Connected successfully to server');
+   
+       await main(client);
+     } finally {
+       // Ensures that the client will close when you finish/error
+       await client.close();
+     }
+   }
+   
+   // Runs your code
+   run();
+
+Inside the ``aggregate`` method, the first pipeline stage filters out
+all documents except those in the ``'Bronx'`` borough. The
+second stage unwinds the ``categories`` field, which is an array, and
+treats each item in the array as a separate document. The third stage
+groups the documents by category and adds up the number of matching
+Bronx restaurants.
+
+Single Purpose Aggregation Operations
+-------------------------------------
+
+MongoDB provides helper methods for some aggregation functions,
+including ``countDocuments``, ``estimatedDocumentCount``, and ``distinct``.
+
+countDocuments
+^^^^^^^^^^^^^^
+
+The following example demonstrates how to use the ``countDocuments`` method to
+find the total number of documents which have the exact array
+``[ 'Chinese', 'Seafood' ]`` in the ``categories`` field.
+
+.. code-block:: js
+
+   async function main(client) {
+     const collection = client.db('myproject').collection('restaurants');
+
+     const count = await collection.countDocuments({
+       categories: [ 'Chinese', 'Seafood' ]
+     });
+
+     console.log(count);
+   }
+
+estimatedDocumentCount
+^^^^^^^^^^^^^^^^^^^^^^
+
+If you want a quick, thought not necessarily accurate, count of the
+number of documents in a collection based on collection metadata, you
+can use the method ``estimateDocumentCount``:
+
+.. code-block:: js
+
+   async function main(client) {
+     const collection = client.db('myproject').collection('restaurants');
+
+     const count = await collection.estimatedDocumentCount();
+
+     console.log(count);
+   }
+
+
+Distinct
+^^^^^^^^
+
+The ``distinct`` helper method eliminates results which contain
+values and returns one record for each unique value. It uses
+the :manual:`distinct </reference/command/distinct/>` command.
```

Comment:
why are we talking about the `distinct` command in the middle of our aggregation tutorial? Honestly, I'm inclined to drop this whole tutorial, I'm sure we have official docs that describe aggregation far better. Unless this is updated to include a bunch of examples of using aggregation specifically with the node driver, I don't find value in this document.

Example of making this more useful would be:
  - basic collection aggregation
  - collection aggregation with $out (write operation)
  - collection aggregation with $merge (write operation)
  - database-level aggregation

I don't even see why it's useful for us to tell people that `estimagedDocumentCount` is actually an aggregation under the hood. If we're doing that, shouldn't we also mention that `watch` is an aggregation? I don't think anyone really cares.

Reply:
I was porting over what was originally in the document. There used to be sections about `count`, `group` and `distinct`. Since `group` is deprecated, I removed it, and I split up `count` into `countDocuments` and `estimatedDocumentCount`. I can delete them if you want. Though it might benefit us in the future to have a quick guide on the differences between `estimatedDocumentCount` and `countDocuments`?

Reply:
Absolutely. I’m not questioning the benefit of documentation between the two count methods, rather that they are in our aggregation documentation